[RFC] Add expand() call to remote_plugins_manifest

skeate opened the pull request:
According to the vim helpfile:

> fnamemodify({fname}, {mods})
>    ...
>    Note: Environment variables don't work in {fname}, use
>    expand() first then.

So this causes issues if your `$MYVIMRC` contains environment variables
(e.g. `$XDG_CONFIG_HOME`)

justinmk commented on runtime/autoload/remote/host.vim:
```diff
@@ -1,6 +1,6 @@
 let s:hosts = {}
 let s:plugin_patterns = {}
-let s:remote_plugins_manifest = fnamemodify($MYVIMRC, ':p:h')
+let s:remote_plugins_manifest = fnamemodify(expand($MYVIMRC), ':h')
```
Always pass `1` to 2nd arg of `expand()`: `expand($MYVIMRC, 1)`

skeate replied:
Updated

skeate committed:
Add expand() call to remote_plugins_manifest

According to the vim helpfile:

> fnamemodify({fname}, {mods})
>    ...
>    Note: Environment variables don't work in {fname}, use
>    expand() first then.

So this causes issues if your $MYVIMRC contains environment variables
(e.g. $XDG_CONFIG_HOME)

justinmk commented:
Merged. Thanks!
